themes/wporg-translate-events-2024/functions.php

```diff
@@ -0,0 +1,145 @@
+				),
+				array(
+					'label' => esc_html__( 'Requests', 'wporg-plugins' ),
+					'url'   => 'https://make.wordpress.org/polyglots/?resolved=unresolved',
+				),
+				array(
+					'label' => esc_html__( 'Weekly Chats', 'wporg-plugins' ),
+					'url'   => 'https://make.wordpress.org/polyglots/category/weekly-chats/',
+				),
+				array(
+					'label' => esc_html__( 'Translate', 'wporg-plugins' ),
+					'url'   => 'https://translate.wordpress.org/',
+				),
+				array(
+					'label' => esc_html__( 'Handbook', 'wporg-plugins' ),
+					'url'   => 'https://make.wordpress.org/polyglots/handbook/',
+				),
+			),
+		);
+	}
+);
+
+// Declare the base breadcrumbs, which apply to all pages.
+// Pages can add additional levels of breadcrumbs.
+add_filter(
+	'wporg_block_site_breadcrumbs',
+	function (): array {
+		return array(
+			array(
+				'url'   => home_url(),
+				'title' => __( 'Home', 'wporg-translate-events-2024' ),
+			),
+			array(
+				'url'   => Urls::events_home(),
+				'title' => __( 'Events', 'wporg-translate-events-2024' ),
+			),
+		);
+	}
+);
+
+/**
+ * Add social tags to the head of the page.
+ *
+ * @param string $html_title       The title of the page.
+ * @param string $url              The URL of the page.
+ * @param string $html_description The description of the page.
+ * @param string $image_url        The URL of the image to use.
+ *
+ * @return void
+ */
+function add_social_tags( string $html_title, string $url, string $html_description, string $image_url ) {
+	$meta_tags = array(
+		'name'     => array(
+			'twitter:card'        => 'summary',
+			'twitter:site'        => '@WordPress',
+			'twitter:title'       => esc_attr( $html_title ),
+			'twitter:description' => esc_attr( $html_description ),
+			'twitter:creator'     => '@WordPress',
+			'twitter:image'       => esc_url( $image_url ),
+			'twitter:image:alt'   => esc_attr( $html_title ),
+		),
+		'property' => array(
+			'og:url'              => esc_url( $url ),
+			'og:title'            => esc_attr( $html_title ),
+			'og:description'      => esc_attr( $html_description ),
+			'og:site_name'        => esc_attr( get_bloginfo() ),
+			'og:image:url'        => esc_url( $image_url ),
+			'og:image:secure_url' => esc_url( $image_url ),
+			'og:image:type'       => 'image/png',
+			'og:image:width'      => '1200',
+			'og:image:height'     => '675',
+			'og:image:alt'        => esc_attr( $html_title ),
+		),
+	);
+
+	foreach ( $meta_tags as $name => $content ) {
+		foreach ( $content as $key => $value ) {
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			echo '<meta ' . esc_attr( $name ) . '="' . esc_attr( $key ) . '" content="' . esc_attr( $value ) . '" />' . "\n";
+		}
+	}
+}
+
+function render_header( string $title ): void {
+	$json = wp_json_encode( array( 'title' => $title ) );
+	// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	echo "<!-- wp:wporg-translate-events-2024/header $json /-->";
+}
```

Comment:
I am not sure I am very happy with such a function. If we want to make it easier, we could do something more generic like this:
```php
<?php block_with_attributes( 'wp:wporg-translate-events-2024/header', array( 'title' => 'My Events' ) );
```

```suggestion
function block_with_attributes( string $block, array $attributes ): void {
	$json = wp_json_encode( $attributes );
	// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
	echo "<!-- $block $json /-->";
}
```

Reply:
I've pushed a commit that removed that function. If in the future we end up requiring such a function, I agree with your suggestion of having a generic  `block_with_attributes()`.

Reply:
Not a big deal in this context, especially because you've removed the function for now, however: it's worth noting that taking the output of `wp_json_encode`/`json_encode` directly into an HTML context like this shouldn't be assumed to be safe. There are a variety of flags that can passed along in the second argument to ensure that the encoded JSON doesn't break in that context. Some combination of the following bitwise flags would do it: `JSON_HEX_QUOT|JSON_HEX_TAG|JSON_HEX_AMP|JSON_HEX_APOS`.

Reply:
Great point about not assuming the data to be safe. Before we use this theme in production, I will make sure that all `wp_json_encode`/`json_encode` calls are safe. Thanks!